### PR TITLE
Post comments to comments route for T16

### LINKED ISSUE #255: T16 comment to be sent to comment route.

### PR DESCRIPTION
**Pull request by olivier-vadiaval**
Post comments to comments route for T16
Add small fix for two comments likes tests

I didn't create an adapter since this change looks a bit trivial imo. But, if there are more changes to make to adapt to t16, I can refactor it.

Fix #255

### CODE REVIEW

**Comment by JawadHossain** [comment]
Getting a 500 error. Logs:
INFO socialDistribution.requests: API POST request to https://i-connect.herokuapp.com/service/author/74545527-87ef-4fee-bd41-17bda554b818/posts/c25b8486-5cc6-4dd2-b6a6-6c4bfd56779e/comments/ and received 500
[01/Dec/2021 18:33:14] "POST /app/author/9f3d3185-cd3b-4ca0-8c18-e0e81bec0dd3/posts/7795a868-cafb-4469-b3f0-eda38032b962/comments/ HTTP/1.1" 302 0

Can you pls check?

**Comment by debecerra** [comment]
Didn't run the code myself, but I know T16 does not use the `object` field like we do. We may need to create an extra field `id` with the same value as `object` when sending to them. Might be the cause of the issue Jawad mentioned.

**Comment by debecerra** [comment]
https://i-connect-doc.herokuapp.com/docs/#/comment/authorAUTHOR_IDPostsPOST_IDCommentsPOST

**Comment by olivier-vadiaval** [comment]
> Getting a 500 error. Logs: INFO socialDistribution.requests: API POST request to https://i-connect.herokuapp.com/service/author/74545527-87ef-4fee-bd41-17bda554b818/posts/c25b8486-5cc6-4dd2-b6a6-6c4bfd56779e/comments/ and received 500 [01/Dec/2021 18:33:14] "POST /app/author/9f3d3185-cd3b-4ca0-8c18-e0e81bec0dd3/posts/7795a868-cafb-4469-b3f0-eda38032b962/comments/ HTTP/1.1" 302 0
> 
> Can you pls check?

When I checked it earlier before the demo, I was getting a 200 OK. I'll try it again and take note of Diego's comment.
Thank you! ♥

**Review by tylerelston — APPROVED** [review]

File: code/socialDistribution/views.py [hunk 1]
```diff
@@ -842,7 +843,12 @@ def post_comment(request, author_id, post_id):
                     "object": post.public_id
                 }
 
-                request_url = f'{post.author.strip("/")}/inbox/'
+                if url_parser.get_host(post.public_id) == REMOTE_NODES["t16"]:
```
Comment (tylerelston) [inline code/socialDistribution/views.py:846]: Thoughts on making this the default URL, not just for T16? It looks like T08 and T11 also use the /comments endpoint for posting new comments.

It looks like it's just our group that posts comments to /inbox

https://github.com/DecKirin/404_group_project_21F/wiki/API-documentation#comments

https://github.com/CMPUT404-F21T0/CMPUT404-Project-BetterSocial/wiki/API-Documentation#comments

**Review by tylerelston — DISMISSED** [review]

File: code/socialDistribution/views.py [hunk 2]
```diff
@@ -841,7 +842,18 @@ def post_comment(request, author_id, post_id):
...
 
-                request_url = f'{post.author.strip("/")}/inbox/'
+                post_host = url_parser.get_host(post.public_id)
+                if post_host != HOST:
+                    request_url = post.public_id.strip("/") + '/comments/'
+                    
+                    if post_host == REMOTE_NODES["t16"]:
+                        data["id"] = post.public_id
+
+                    elif post_host == REMOTE_NODES["t11"]:
```
Comment (JawadHossain) [inline code/socialDistribution/views.py:852]: can you pls also add "t20" as they also expect it at //service/author/{author_id}/posts/{post_id}/comments

**Review by JawadHossain — APPROVED** [review]
looks great, thanks!

File: code/socialDistribution/views.py (hunk 2 above)
Comment (olivier-vadiaval) [reply code/socialDistribution/views.py:852]: Sounds good
Will add them
Thanks! 🙏

Comment (olivier-vadiaval) [reply code/socialDistribution/views.py]: Actually, the code was sending it to the comments route for every team. One issue that I found though was that we need put the post's public_id in an id field of our payload for every team except team 11, which expects it inside an api_url field for some reason.

**Review by debecerra — APPROVED** [review]

**Comment by JawadHossain** [comment]
t20 just messaged and requested to send comments to their inbox. sorry for the confusion

**Comment by olivier-vadiaval** [comment]
> @olivier-vadiaval t20 just messaged and requested to send comments to their inbox. sorry for the confusion

No worries 😉 I'll make the change

**Review by JawadHossain — APPROVED** [review]

**Pull request merged**